ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/ForkIdManager.java

```diff
@@ -40,23 +41,25 @@
   private final List<ForkId> forkAndHashList;
 
   private final List<Predicate<ForkId>> forkIDCheckers;
+  private final List<Long> forks;
+  private final LongSupplier chainHeadSupplier;
+  private long lastHead;
+  private ForkId lastComputedForkId;
 
   public ForkIdManager(final Blockchain blockchain, final List<Long> forks) {
     checkNotNull(blockchain);
     checkNotNull(forks);
+    this.chainHeadSupplier = blockchain::getChainHeadBlockNumber;
     this.genesisHash = blockchain.getGenesisBlock().getHash();
     this.forkAndHashList = new ArrayList<>();
+    this.forks =
+        forks.stream()
+            .filter(fork -> fork > 0)
+            .distinct()
+            .sorted()
+            .collect(Collectors.toUnmodifiableList());
     final Predicate<ForkId> legacyForkIdChecker =
-        createForkIDChecker(
-            blockchain,
-            genesisHash,
-            forks,
-            fs ->
-                fs.stream()
-                    .filter(fork -> fork > 0)
-                    .distinct()
-                    .collect(Collectors.toUnmodifiableList()),
-            forkAndHashList);
+        createForkIDChecker(blockchain, genesisHash, forks, fs -> forks, forkAndHashList);
```

Comment:
If this isn't used anywhere else, we should probably move it into the else clause.

Reply:
Non relevant anymore.